### PR TITLE
implement entity and alias create, update, delete support

### PR DESCRIPTION
**Pull request by dwelch0**
TICKETS: 
* https://issues.redhat.com/browse/APPSRE-4656
* https://issues.redhat.com/browse/APPSRE-4657

### CODE REVIEW

**Comment by dwelch0** [comment]
Dependency: https://github.com/app-sre/qontract-schemas/pull/106

**Comment by dwelch0** [comment]
sample definition used during dev:
`---
$schema: /vault-config/entity-1.yml

labels:
  service: vault.fedramp.devshift.net

name: "drewwelch"
type: "entity"
policies:
- mt-sre-policy
- platform-feedback-policy
metadata:
  organization: "RED HAT"
  team: appsre
  position: swe-sre
aliases:
- name: dwelch0
  auth_type: github
  custom_metadata:
    description: "Professional GitHub Account"
- name: dwelch
  auth_type: oidc
  custom_metadata:
    description: "RH SSO Account"
    env: commercial
`

File: pkg/vault/reconcile.go [hunk 1]
```diff
@@ -194,22 +186,3 @@ func DataInSecret(data map[string]interface{}, path string) bool {
 	}
 	return true
 }
-
-// Yaml unmarshal limitation causes nested options objects to be decode as strings with json format
-// ex: `{"foo": "bar"}`
-// UnmarshalJsonObj performs unmarshal of jsons strings
-func UnmarshalJsonObj(key string, obj interface{}) (map[string]interface{}, error) {
```
Comment (dwelch0) [inline pkg/vault/reconcile.go:201]: This function was originally implemented with work to support oidc roles. Once use case for this functionality also arose within entity implementation, I elected to extract definition from role package and generalize within pkg/utils

**Review by TGPSKI — CHANGES_REQUESTED** [review]
Drew,

This is awesome work and I'm super happy to have this locked in.

I'm looking for something a bit different on the data modeling side. Having everything in a flat file is great and we may have to use it for SRR deadline. All of the relationships we are seeking to model in Vault already exist in app-interface in another location.

Ideal situation:

* There is no entity schema. Entities are automatically created when a user file is added to app-interface, following the user-1.yml schema. (https://github.com/app-sre/qontract-schemas/blob/main/schemas/access/user-1.yml)
* A further improvement would be, only add an entity / entity alias if the user is associated with a role which provides vault permissions. This is not strictly required, but is an optional feature. Making a entity / entity alias pair for every user is acceptable.

Example data:

* user file tpate.yml exists (https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/teams/app-sre/users/tpate.yml)
* tpate.yml is part of the app-sre.yml role (/teams/app-sre/roles/app-sre.yml)
* the app-sre.yml role provides permissions to vault (https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/teams/app-sre/roles/app-sre.yml#L16)
* because the user file exists, and the role associated with the user file provides permissions to vault, then create an entity and entity alias for the user. do not track these in a separate yaml file. infer their presence based on the presence of the user file, the role file, and the permissions file

Let me know if this makes sense.

**Pull request closed without merging**